### PR TITLE
[update] new hints in the example of github-pages.md

### PR DESCRIPTION
**Pull request by NatLee**
There are some issues with this document.

- Github do not support the name of token starting with `GITHUB_`.
  Check the issue here -> https://github.com/github/docs/issues/12094

- The name of `token` in the `pages.yml` is ambiguous. User may check the usage of actions.

## Check List

**Please read and check followings before submitting a PR.**

- [ ] I want to publish my theme on Hexo official website.
  - [ ] I have read the [theme publishing doc](https://hexo.io/docs/themes#Publishing).
  - [ ] `name` is unique.
  - [ ] `link` URL is correct.
  - [ ] `preview` URL is correct.
  - [ ] `preview` URL web site is rendered correctly.
  - [ ] Add a screenshot to `source/themes/screenshots`.
  - [ ] Screenshot filename is same as value of `name`.
  - [ ] Screenshot size is `800 * 500`.
  - [ ] Screenshot file format is `png`.
- [ ] I want to publish my plugin on Hexo official website.
  - [ ] I have read the [plugin publishing doc](https://hexo.io/docs/plugins#Publishing).
  - [ ] `name` is unique.
  - [ ] `link` URL is correct.
- [x] Others (Update, fix, translation, etc...)
  - Languages:
  - [x] `en` English
  - [ ] `ru` Russian
  - [x] `zh-cn` simplified Chinese
  - [x] `zh-tw` traditional Chinese

> **Note**
>
> No need to translate `ko`, `pt-br`, `th` languages if you submit a PR for documentation.

> **Warning**
>
> `ko`, `pt-br`, `th` languages document are inactive. Please, submit issue if you want to continuous contribution to them.

### CODE REVIEW

File: source/zh-tw/docs/github-pages.md [hunk 1]
```diff
@@ -51,7 +52,8 @@ jobs:
       - name: Deploy
         uses: peaceiris/actions-gh-pages@v3
         with:
-          github_token: ${{ secrets.GITHUB_TOKEN }}
+          # Please check the supported token here: https://github.com/peaceiris/actions-gh-pages#supported-tokens
+          github_token: ${{ secrets.DEPLOY_TOKEN }}
```
Comment (renbaoshuo) [inline source/zh-tw/docs/github-pages.md:56]: .

File: source/zh-tw/docs/github-pages.md [hunk 2]
```diff
@@ -30,7 +30,8 @@ jobs:
     steps:
       - uses: actions/checkout@v3
         with:
-          token: ${{ secrets.GITHUB_TOKEN }}
+          # If you have other token like `ssh-key`, please check the usage of this action: https://github.com/actions/checkout#usage
+          token: ${{ secrets.DEPLOY_TOKEN }}
```
Comment (renbaoshuo) [inline source/zh-tw/docs/github-pages.md:34]: The `GITHUB_TOKEN` is a special token that is created automatically for each workflow run.

https://docs.github.com/en/actions/security-guides/automatic-token-authentication#about-the-github_token-secret

**Review by renbaoshuo — CHANGES_REQUESTED** [review]

**Comment by renbaoshuo** [comment]
The `GITHUB_TOKEN` is a special token that is created automatically for each workflow run.

https://docs.github.com/en/actions/security-guides/automatic-token-authentication#about-the-github_token-secret

**Comment by renbaoshuo** [comment]
Sorry, I clicked the "Close Pull Request" button by mistake.

**Comment by stevenjoezhang** [comment]
See also https://github.com/github/docs/pull/12095#issuecomment-974285693

**Comment by NatLee** [comment]
@renbaoshuo Thanks for your fast response!

I have trid to use the token starts with `GITHUB_` in actions.

image

But, I got a failed message below

image

In my opion, we need to change the name of token in the document or they may confuse the user.

**Comment by stevenjoezhang** [comment]
The token named `secrets.GITHUB_TOKEN` is created automatically. You don't need to create it.
You just need to follow the steps in [github-pages.md](https://github.com/hexojs/site/pull/1979#diff-d396a1809441d09cb30c2d7eaf955c7f215f72f0d1630071d9fd9cdfecbbd049), it should not throw any errors.

**Comment by NatLee** [comment]
@stevenjoezhang Thanks!

I followed the step here https://hexo.io/zh-tw/docs/github-pages.html and created a new repo to test the workflows.

In the stage of deployment, it showed error of 403:

image

**Comment by stevenjoezhang** [comment]
截屏2023-02-15 15 55 50

It is working in my repository... Maybe you can refer to https://github.com/peaceiris/actions-gh-pages/issues/744

**Review by SukkaW — CHANGES_REQUESTED** [review]
Just blocking the pull request as the change requires further discussion/investigation.

**Comment by NatLee** [comment]
@stevenjoezhang Thanks!

In that reference, we need to switch the setting in actions.

image


I find there is another way to give writting permission. -> https://github.com/peaceiris/actions-gh-pages/pull/753

Just add permissions in the `.yml`.

```
jobs:
  pages:
    runs-on: ubuntu-latest
    permissions:
      contents: write
    ...
```

**Review by renbaoshuo — APPROVED** [review]

**Review by stevenjoezhang — APPROVED** [review]

**Review by SukkaW — APPROVED** [review]

**Pull request merged**